### PR TITLE
Electrs integration for Bitcoin chain

### PR DESCRIPTION
**Pull request by nkuba**
This PR implements Bitcoin package integration with Electrs API.

TODO:
- [ ] command flags for config properties
- [ ] tests
- [ ] PR description

Refs https://github.com/keep-network/keep-core/pull/3384
Refs https://github.com/keep-network/keep-core/issues/3369

### CODE REVIEW

File: pkg/bitcoin/chain.go [hunk 1]
```diff
@@ -20,12 +20,12 @@ type Chain interface {
 	// rejected eventually.
 	BroadcastTransaction(transaction *Transaction) error
 
-	// GetCurrentBlockNumber gets the number of the current block. If the
-	// current block was not determined, this function returns an error.
-	GetCurrentBlockNumber() (uint, error)
+	// GetLatestBlockHeight gets the height of the latest block (tip). If the
+	// latest block was not determined, this function returns an error.
+	GetLatestBlockHeight() (uint, error)
```
Comment (nkuba) [inline pkg/bitcoin/chain.go:25]: @lukasz-zimnoch I hope you're fine with this rename.

The function expects the "chain tip", so the most recent mined block. I propose we use "latest", as "current" may be confusing and suggest that the block is being currently mined.

Also, I replaced "block number" with "block height", as it seems to be widely used in the Bitcoin ecosystem.

File: pkg/bitcoin/electrs/block.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package electrs
+
+import (
+	"fmt"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+)
+
+type blockHeader struct {
```
Comment (tomaszslabon) [inline pkg/bitcoin/electrs/block.go:9]: Let's not forget about descriptions of all the functions and types.

File: pkg/bitcoin/electrs/block.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+func (b *blockHeader) convert() (bitcoin.BlockHeader, error) {
+	result := bitcoin.BlockHeader{
+		Version: b.Version,
+		Time:    b.Timestamp,
+		Bits:    b.Bits,
+		Nonce:   b.Nonce,
+	}
+
+	previousBlockHeaderHash, err := bitcoin.NewHashFromString(b.PreviousBlockHash, bitcoin.ReversedByteOrder)
```
Comment (tomaszslabon) [inline pkg/bitcoin/electrs/block.go:26]: Could we split this lengthy line?

File: pkg/bitcoin/electrs/electrs.go [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+	retryTimeout time.Duration
+}
+
+type httpClient interface {
+	Get(url string) (*http.Response, error)
+	Post(url string, contentType string, body io.Reader) (*http.Response, error)
+}
+
+// Connect initializes handle with provided Config.
+func Connect(config Config) (bitcoin.Chain, error) {
```
Comment (tomaszslabon) [inline pkg/bitcoin/electrs/electrs.go:45]: `Connect` won't return an error even if the URL provided in the config is wrong.
We will only learn about the wrong URL when we try to use the Bitcoin chain for the first time.
I would consider adding a connection test before we leave `Connect`, we did that such test when connecting to btc client in [the old relay maintainer](https://github.com/keep-network/tbtc/blob/d18ef9aec7656f0ec7d317ece3e3d5c7aca92cda/relay/pkg/btc/remote.go#L45)

**Comment by nkuba** [comment]
Closing in favour of https://github.com/keep-network/keep-core/pull/3406

**Pull request closed without merging**